diff --git a/sources/ade/CMakeLists.txt b/sources/ade/CMakeLists.txt
index 2d1dd20..8e19233 100644
--- a/sources/ade/CMakeLists.txt
+++ b/sources/ade/CMakeLists.txt
@@ -14,7 +14,8 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL GNU)
     set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra -Wconversion -Wshadow -Wno-error=cpp -Wformat -Wformat-security" )
 endif()
 
-add_library( ${PROJECT_NAME} STATIC ${include} ${sources} )
+add_library( ${PROJECT_NAME} SHARED ${include} ${sources} )
+set_target_properties( ${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION 0 )
 target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
 set_target_properties( ${PROJECT_NAME} PROPERTIES POSITION_INDEPENDENT_CODE True)
 
@@ -49,12 +49,12 @@ endif()
 
 install(TARGETS ade COMPONENT dev
         EXPORT adeTargets
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib
-        RUNTIME DESTINATION lib
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_LIBDIR}"
         INCLUDES DESTINATION include)
 
-install(EXPORT adeTargets DESTINATION share/ade COMPONENT dev)
+install(EXPORT adeTargets DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/ade" COMPONENT dev)
 
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/"
         DESTINATION include COMPONENT dev
@@ -70,10 +70,10 @@
 configure_package_config_file(
         "${CMAKE_CURRENT_LIST_DIR}/cmake/adeConfig.cmake.in"
         "${CMAKE_BINARY_DIR}/adeConfig.cmake"
-        INSTALL_DESTINATION "share/ade"
+        INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/ade"
         NO_SET_AND_CHECK_MACRO
         NO_CHECK_REQUIRED_COMPONENTS_MACRO)
 
 install(FILES "${CMAKE_BINARY_DIR}/adeConfig.cmake"
         "${CMAKE_BINARY_DIR}/adeConfigVersion.cmake"
-        DESTINATION "share/ade" COMPONENT dev)
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/ade" COMPONENT dev)
